src/safeposix/syscalls/sys_calls.rs

```diff
@@ -219,12 +270,16 @@ impl Cage {
             panic!("We changed from a directory that was not a directory in chdir!");
         }
 
-        // we grab the parent cages main threads sigset and store it at 0
-        // we do this because we haven't established a thread for the cage yet, and dont have a threadid to store it at
-        // this way the child can initialize the sigset properly when it establishes its own mainthreadid
+
+        // We clone the parent cage's main threads and store them and index 0 
+        // This is done since there isn't a thread established for the child Cage object yet - 
+        // And there is no threadId to store it at. 
+        // The child Cage object can then initialize and store the sigset appropriately when it establishes its own 
+        // main thread id.
         let newsigset = interface::RustHashMap::new();
         if !interface::RUSTPOSIX_TESTSUITE.load(interface::RustAtomicOrdering::Relaxed) {
```

Comment:
What is the use of this code section here, can you explain a bit about it?

Reply:
which code section? the one where it checks for the test suite?

Reply:
Yes.

Reply:
The comments below the if condition explains why we are checking for the test suite